### PR TITLE
[#38] Add some partial functions Inspections

### PR DESCRIPTION
**Pull request by vrom911**
Part of #38

### CODE REVIEW

File: src/Stan/Analysis/Analyser.hs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
```
Comment (chshersh) [inline src/Stan/Analysis/Analyser.hs:5]: A few words about this module :slightly_smiling_face: 
```suggestion

Analysing function with an 'Id' of the corresponding 'Inspection'.
```

File: src/Stan/Inspection/Partial.hs [hunk 2]
```diff
@@ -8,9 +8,22 @@ Contains all 'Inspection's for partial functions.
 
 module Stan.Inspection.Partial
     ( -- * Partial 'Inspection's
+      -- *** Partial 'GHC.List.head'
       stan0001
     , stan0001Inspection
     , stan0001Meta
+      -- *** Partial 'GHC.List.tail'
```
Comment (chshersh) [inline src/Stan/Inspection/Partial.hs:15]: Top idea! :+1:

File: src/Stan/Inspection/Partial.hs [hunk 3]
```diff
@@ -43,23 +59,67 @@ mkPartialInspection insId NameMeta{..} = Inspection
...
--- | Corresponding 'NameMeta' for 'stan0001' — partial 'GHC.List.head' @STAN-0001@.
-stan0001Meta :: NameMeta
-stan0001Meta = NameMeta
-    { nameMetaName       = "head"
+mkBaseListMeta :: Text -> NameMeta
+mkBaseListMeta funName = NameMeta
+    { nameMetaName       = funName
     , nameMetaPackage    = "base"
     , nameMetaModuleName = "GHC.List"
     }
```
Comment (chshersh) [inline src/Stan/Inspection/Partial.hs:78]: I will need this function for infinite checks as well... Should we create `Stan.NameMeta` or should we move into `Stan.Inspections.Common` or something else?

File: test/Test/Stan/Analysis/Partial.hs [hunk 4]
```diff
@@ -3,35 +3,53 @@ module Test.Stan.Analysis.Partial
...
-        foundHeadObservation `shouldBe` Just expectedHeadObservation
+analysisHeadSpec analysis = describe "Partial functions" $ do
+    it "STAN-0001: finds usage of 'base/head'" $
+        foundObservation analysis 1 stan0001 7 12 16
+    it "STAN-0002: finds usage of 'base/tail'" $
+        foundObservation analysis 1 stan0002 10 12 16
+    it "STAN-0003: finds usage of 'base/init'" $
+        foundObservation analysis 1 stan0003 13 12 16
+    it "STAN-0004: finds usage of 'base/last'" $
+        foundObservation analysis 1 stan0004 16 12 16
```
Comment (chshersh) [inline test/Test/Stan/Analysis/Partial.hs:24]: Juicy tests! :+1: :heart:

File: test/Test/Stan/Inspection.hs [hunk 5]
```diff
@@ -12,9 +12,16 @@ import qualified Stan.Inspection.Partial as Stan
...
-    describe "Partial" $
+    describe "Partial" $ do
         it "STAN-0001 should be partial head" $
             getInspectionById Stan.stan0001 `shouldBe` Stan.stan0001Inspection
+        it "STAN-0002 should be partial tail" $
+            getInspectionById Stan.stan0002 `shouldBe` Stan.stan0002Inspection
+        it "STAN-0003 should be partial init" $
+            getInspectionById Stan.stan0003 `shouldBe` Stan.stan0003Inspection
+        it "STAN-0004 should be partial last" $
+            getInspectionById Stan.stan0004 `shouldBe` Stan.stan0004Inspection
```
Comment (chshersh) [inline test/Test/Stan/Inspection.hs:23]: I wonder, if we can automate at least this set of tests :thinking: Something like:

```haskell
forM_ (zip partialInspections partialInspectionsIds) $ \(ins, insId) ->
    it (unId insId <> ": " <> inspectionName ins) $
        getInspectionById insId `shouldBe` ins
```

File: src/Stan/Inspection/Partial.hs (hunk 3 above)
Comment (vrom911) [reply src/Stan/Inspection/Partial.hs:78]: `Stan.NameMeta` sounds like the right place. I can do this :+1:

**Pull request merged**